goon/browserassets/js/browserOutput.js

```diff
@@ -166,30 +151,76 @@ function emojiparse(el) {
 	}
 }
 
-// Colorizes the highlight spans
-function setHighlightColor(match) {
-	match.style.background = opts.highlightColor
+// Escape any special regex characters in a given string, so that one can match it literally.
+function escapeRegex(input){
+	return input.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+}
+
+// Recursively highlights all text matching `term` in text nodes reachable from the given root `element`.
+// Returns the last (right-most, DOM-wise) element that was analyzed and potentially highglighted.
+function highlightRecursively(element, term) {
+	var regex = new RegExp(term, "i");
+	// if that's a text node - wrap found words in a highlighter span.
+	if (element.nodeType == Node.TEXT_NODE) {
+		// The underlying text element would (likely) be destroyed, and we'll be left with
+		// a collection of smaller text nodes and highlighter spans. They'd be stored in this array.
+		var newElements = [];
+		var lastStop = 0;  // The end of the last found match.
+		// The loop is ugly.
+		// There are reasons for it, namely: BYOND does not support matchAll regex method, and the `regexp.exec()` which is
```

Comment:
What exactly do you mean by "BYOND" here? We're dealing with an IE browser instance, right?